templates/service.yaml

```diff
@@ -30,6 +30,12 @@ spec:
 {{- if (and (eq .Values.service.type "NodePort") (not (empty .Values.service.nodePort))) }}
       nodePort: {{ .Values.service.nodePort }}
 {{- end }}
+      {{- if .Values.serviceMonitor.enabled }}
+    - port: {{ (split ':' .Values.configData.http.debug.addr)._1 }}
```

Comment:
If we're discarding the IP address portion of `configData.http.debug.addr`, let's just rename `addr` to `port` and have that expect the port alone. Then, this line could just be `- port: {{ .Values.configData.http.debug.port }}`.